### DISCUSSION TITLE
How can I find Vuetify types for TypeScript support?

### DISCUSSION THREAD

**Discussion by strokesws** [vuetifyjs/vuetify#13721] [Q&A] [answered]
Hi! I'm trying to find Vuetify types while building my application with TypeScript, I'm using Vue 2. How can I find them so I can import into my code? Some examples:

VCombobox - Where can I find the interface of the VCombobox `items` so that when `changeValue` is called the first parameter can be defined
```
<template>
  <v-combobox
    label="My comboboc"
    :items="items"
    @keydown="onChangeComboboxItem"
  />
</template>

<script lang="ts">
  import { defineComponent } from '@vue/composition-api';

  export default defineComponent({
    setup() {
      const items = []; // some objects inside of this array

      const onChangeComboboxItem = (comboboxItem) => {
        console.log(comboboxItem);
      };

      return {
        items,
        onChangeJobTemplate,
      };
    },
  });
</script>
```

VDataTable - Where can I find the interface of the first and second param when `onDbClickRowItem`?
```
<template>
  <v-data-table
    :headers="headers"
    :items="items"
    @dbclick:row="onDbClickRowItem"
  />
</template>

<script lang="ts">
  import { defineComponent } from '@vue/composition-api';

  export default defineComponent({
    setup() {
      const headers = []; // my headers
      const items = []; // my items

      const onDbClickRowItem = (_, rowItem) => { 
        console.log(_)
        console.log(rowItem)
      };

      return {
        items,
        onChangeJobTemplate,
      };
    },
  });
</script>
```

**Comment by KaelWD** [comment, marked as answer]
> Where can I find the interface of the VCombobox items

You can't, it's `any[]` because it accepts pretty much anything. That type should be based on the type of items you're passing in. 

> first and second param when `onDbClickRowItem`

The first is `MouseEvent`, the second is `DataTableItemProps` from `vuetify/types`

**Reply by strokesws** [reply]
Gotcha! So I could possibly do this in my function definition:

```
const onDbClickRowItem = (
  _event: MouseEvent,
  rowItem: Omit<DataTableItemProps, 'item'> & {
    item: IMyItemInterface;
  },
) => {
  console.log(_event)
  console.log(rowItem)
};
```

I guess? 😂 

Would it be useful maybe to add `DataTableItemProps` to the documentation? Or is there a specific reason for it not being there?
